pkg/util/validation/exporter/exporter.go

```diff
@@ -167,60 +164,47 @@ func (oe *OverridesExporter) RingHandler(w http.ResponseWriter, req *http.Reques
 	util.WriteHTMLResponse(w, ringDisabledPage)
 }
 
-// ownsTenant determines whether this overrides-exporter instance owns the given
-// tenant. If the ring is disabled, it assumes this instance has ownership of all
-// tenants. If the ring is enabled, it uses it to check ownership of the given
-// tenant.
-func (oe *OverridesExporter) ownsTenant(tenantID string) bool {
+// isLeader determines whether this overrides-exporter instance is the leader
+// replica that exports all limit metrics. If the ring is disabled, leadership is
+// assumed. If the ring is enabled, it is used to determine which ring member is
+// the leader replica.
+func (oe *OverridesExporter) isLeader() bool {
 	if oe.ring == nil {
-		// If sharding is not enabled, every instance exports metrics for every tenant
+		// If the ring is not enabled, export all metrics
 		return true
 	}
-	owned, err := oe.ring.Owns(tenantID)
+	if oe.Service.State() != services.Running {
+		// We haven't finished startup yet, likely waiting for ring stability.
+		return false
+	}
+	isLeaderNow, err := oe.ring.isLeader()
 	if err != nil {
-		level.Warn(oe.logger).Log("msg", "overrides-exporter failed to determine tenant ownership", "err", err.Error())
-		// if there was an error establishing ownership using the ring, err on the safe
-		// side and assume this instance owns the tenant
+		level.Warn(oe.logger).Log("msg", "overrides-exporter failed to determine ring leader", "err", err.Error())
+		// If there was an error establishing ownership using the ring, assume leadership
+		// and export metrics anyway.
 		return true
```

Comment:
I have mixed feeling about default to `true` here. I'm wondering if we should be more conservative on the exported metrics, log warning but return `false`. If there's a temporarily issue with the ring, then all replicas will expose the limits metrics, which could be quite problematic.

Reply:
You're probably right, given that probably a super high uptime for these metrics isn't crucial, it's better to default to no metrics for transient issues than to suddenly expose them everywhere.